[KT-47] Rubocop fixes

theodoreleebrant opened the pull request:
Various rubocop fixes in the files:

* Bumped Rubocop to 1.11.0 (from 0.52.0)
* Auto-linted fixes by rubocop
* Useless assignments removed
* Changed all Rails.root.join to use slashes
* Modified all 1/2-letter variable names to be more descriptive.

Requesting help for 6 additional offences which I am not sure / afraid to touch:
1. `app/controllers/application_controller.rb:82:5: C: Naming/MemoizedInstanceVariableName: Memoized variable @contest does not match method name load_contest_from_contest_params. Use @load_contest_from_contest_params instead.`
Usage: `@contest ||= Contest.find_by id: params[:contest_id]`
I think this one has to let slide, as it's an [unsafe cop](https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Naming/MemoizedInstanceVariableName). Thoughts?
2. ` Style/StringConcatenation: Prefer string interpolation to string concatenation.`
Usages: (app/controllers/error_controller.rb:8:20, app/helpers/application_helper.rb) 
`regex_string = '(' + whitelist.map { |i| Regexp.escape(i) }.join('|') + ')'`
I'm a little bit scared of touching this one since there is a bunch of stuff that string interpolation is not as straightforward as most fixes.
3. `app/models/concerns/user_contest_scope.rb:91:5: W: Lint/ConstantDefinitionInBlock: Do not define constants this way within a block.`
Usage: `CUTOFF_CERTIFICATE = 1`
Only a warning, I am inclined to let this one be.
4. `app/models/long_submission.rb:49:7: C: Naming/AccessorMethodName: Do not prefix reader method names with get_.`
Usage: `def get_score_text`
Any suggestion what to change this into?
5. `config/initializers/social_media_text.rb:5:13: C: Style/DocumentDynamicEvalDefinition: Add a comment block showing its appearance if interpolated.`
Usage: binding.eval "\"#{self}\""
This one I am not really sure what to do with.

theodoreleebrant committed:
Bump Rubocop to 1.11.0

Install rubocop-rails since rails package is moved
Install rubocop-rake since rake parts is moved
Update obsolete cops
Remove TargetRailsVersion and AllCops Include to check everything using current rubocop
Remove references to Vagrantfile

theodoreleebrant committed:
Initial rubocop fixes

theodoreleebrant committed:
Initial lint using Rubocop

Increase AbcSize to 100
Fix AccessModifierDeclarations

theodoreleebrant committed:
Remove useless assignments via rubocop

theodoreleebrant committed:
Change Rails.root.join to uniformly use slash

theodoreleebrant committed:
Remove rake linting

Additional linter not needed

theodoreleebrant committed:
Tweak Rubocop YAML file

- Enable LineLength
- Exclude application.rb (empty, used for CanCanCan)
- Disable HelperInstanceVariable

theodoreleebrant committed:
Change 1/2-letter variable names

donjar commented:
Hey thanks for working on this.

1. We should probably aim to refactor this as it can be too magical (`@contest` is randomly set by default in the application controller). However for now maybe we can just leave this in first and put a cop ignore in this line?
2. Can just ignore this cop entirely first.
3. If this is the only violation for the cop in the entire codebase maybe we can just move the constant out of the class?
4. Just `score_text` should be fine
5. Let's ignore this cop in this file for the moment. We should refactor this piece of code anyways, see https://ktom.atlassian.net/browse/KT-45.

I will slowly review the files soon. Can you also make some sort of a todo list on rubocop related stuff that we want to work on but is not covered in this PR? For example points 1 and 2 above.

donjar commented on .rubocop.yml:
```diff
@@ -12,14 +12,20 @@ AllCops:
+
 Metrics/AbcSize:
-  Max: 26
+  Max: 100
```
Change to `Enabled: false` instead?

theodoreleebrant replied:
Alright, disabling AbcSize.

donjar commented on app/controllers/sessions_controller.rb:
```diff
@@ -12,7 +12,7 @@ def new
 
   # Life is hard...
-  # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+  # rubocop:disable Metrics/MethodLength
```
Should remove these rubocop disables?

theodoreleebrant replied:
The problem lies in the big else-if block, and it will keep triggering the cop if left there.

theodoreleebrant replied:
Removed.

theodoreleebrant replied:
Current limit is 17 lines, and the method has 54 lines.

donjar commented on app/models/concerns/contest_files.rb:
```diff
@@ -38,24 +38,20 @@ def refresh_results_pdf
   end
 
-  def submissions_location
+  private def submissions_location
```
Why is this not corrected yet the bottom one is corrected?

theodoreleebrant replied:
Huh, that's weird. Fixed.

theodoreleebrant replied:
Ok, this is really weird. After I fixed it, it appears as a rubocop offense.

theodoreleebrant replied:
This issue is quite weird: it doesn't actually appear as a rubocop warning. Nevertheless, fixed the top 2.

donjar commented on lib/tasks/auto_annotate_models.rake:
```diff
@@ -5,49 +5,49 @@
 
-  # rubocop:disable BlockLength
-  task :set_annotation_options do
+  # rubocop:disable Metrics/BlockLength
```
Disable this cop

theodoreleebrant replied:
Sure, disabling Metrics/BlockLength. For this file, it was auto-generated (presumably the people doing rake knows that it breaks rubocop's cops too)

donjar reviewed (commented):
Thanks again for working on this! Left some comments below...

theodoreleebrant committed:
Remove Metrics/AbcSize and Metrics/BlockLength

theodoreleebrant committed:
Change get_score_text to score_text

theodoreleebrant committed:
Remove rubocop annotations in auto_annotate_models

Since Metrics/BlockLength is disabled, this annotation
is no longer needed.

theodoreleebrant committed:
Make explicit Rails/FilePath enforce slash, fix warnings

theodoreleebrant reviewed (commented):
For 3, I'm leaving it there for now since there are a bunch of references to it that requires the constant to be in the class.
Making new ticket for this [KT-60] as it is not trivial and requires some tests

theodoreleebrant committed:
Temporarily disable two cops

Naming/MemoizedInstanceVariableName and Style/DocumentDynamicEvalDefinition
needs significant refactoring and would be associated with new tickets.

Related ticket:
KT-45 for social_media_text.rb

donjar reviewed (approved):
LGTM, thanks for working on this big change! 🚢
